### ISSUE TITLE
Long values are not stored properly

### ISSUE THREAD

**Issue by marcoy** [arangodb/arangodb-java-driver#78]
In our codebase, we generate our own IDs of type `Long`. When we try to store an object with the `Long` ID field, then retrieving it back. The ID value is completely different.

We are using arangodb version 3.1.1 and java driver 4.1.0

I have an object with an `Long` field (e.g. `LongContainer{lval=71711037588448}`). In this case, the value is `71711037588448`. But then when I tried to retrieve it back through different methods, `getDocument`, JSON String, and AQL. The `Long` value in the object I got back was `2263614432`.

The following code is what I used:
```
package com.marcoy;

import com.arangodb.ArangoCollection;
import com.arangodb.ArangoCursor;
import com.arangodb.ArangoDB;
import com.arangodb.ArangoDBException;
import com.arangodb.entity.DocumentCreateEntity;
import com.fasterxml.jackson.databind.DeserializationFeature;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.google.common.base.MoreObjects;

import java.io.IOException;
import java.util.Collections;
import java.util.Objects;

public class LongValueTest {

    public static class LongContainer {
        private Long lval;

        public LongContainer() { }

        public LongContainer(Long lval) {
            this.lval = lval;
        }

        public Long getLval() {
            return lval;
        }

        public void setLval(Long lval) {
            this.lval = lval;
        }

        @Override
        public String toString() {
            return MoreObjects.toStringHelper(this)
                    .add("lval", lval)
                    .toString();
        }

        @Override
        public boolean equals(Object o) {
            if (this == o) return true;
            if (!(o instanceof LongContainer)) return false;
            LongContainer that = (LongContainer) o;
            return Objects.equals(lval, that.lval);
        }

        @Override
        public int hashCode() {
            return Objects.hash(lval);
        }
    }

    private static String query = "FOR v IN MyTest " +
                                  "RETURN v";

    public static void main(String[] args) throws InterruptedException, IOException {
        final ObjectMapper om = new ObjectMapper();
        om.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);

        final ArangoDB arangoDB = new ArangoDB.Builder()
                .user("root")
                .password("")
                .build();

        System.out.println("Using arangodb version: " + arangoDB.getVersion().getVersion());
        // > Using arangodb version: 3.1.1

        try {
            arangoDB.db().createCollection("MyTest");
        } catch (ArangoDBException ignored) {
        }

        final ArangoCollection myTest = arangoDB.db().collection("MyTest");

        myTest.truncate();

        final LongContainer originalObject = new LongContainer(71711037588448L);
//        final LongContainer originalObject = new LongContainer(Integer.MAX_VALUE + 10001L);
        System.out.printf("Original object: %s\n", originalObject);
        // > LongContainer{lval=71711037588448}

        final DocumentCreateEntity<LongContainer> de1 = myTest.insertDocument(originalObject);

        final String retr1 = myTest.getDocument(de1.getKey(), String.class);
        System.out.printf("As JSON string: %s\n", retr1);
        // > As JSON string: {"_key":"2416","_id":"MyTest\/2416","_rev":"_UHGinMS---","lval":2263614432}

        final LongContainer retr2 = myTest.getDocument(de1.getKey(), LongContainer.class);
        System.out.printf("GetDoucment: %s\n", retr2);
        // > GetDoucment: LongContainer{lval=2263614432}

        final ArangoCursor<LongContainer> results = arangoDB.db().query(LongValueTest.query, Collections.emptyMap(), null, LongContainer.class);
        System.out.printf("AQL: %s\n", results.asListRemaining());
        // > AQL: [LongContainer{lval=2263614432}]
    }
}
```

**Comment by mpv1989** [comment]
this looks like the bug from #73 . I fixed it with version 4.1.3.

**Comment by marcoy** [comment]
Upgrading to 4.1.3 fixes the issues. Thank you.
